frontend-react/src/typings/api-codegen.d.ts

```diff
@@ -0,0 +1,150 @@
+type Topic = "FULL_ELR" | "COVID_19" | "MONKEYPOX" | "CSV_TESTS" | "TEST";
+
+interface ReportStreamFilters {
+    jurisdictionalFilter: string[] | undefined;
+    processingModeFilter: string[] | undefined;
+    qualityFilter: string[] | undefined;
+    routingFilter: string[] | undefined;
+    topic: Topic;
+}
+
+type Jurisdiction = "FEDERAL" | "STATE" | "COUNTY";
+
+interface Organization {
```

Comment:
I was Googling to try and figure out the benefit of `.d.ts` over exporting a type/interface from a `.ts` file. Here's what I found:

>If you define a file with an exported interface (that will not output any js), you should do write it in a .d.ts file. In that way, it will be excluded by the compiler when generating modules.

Is my understanding correct in saying "if we move our types/interfaces to type declaration files rather than exporting them, we can cut our static build size down"?

Reply:
All typescript types/interfaces (except enum I think) go away on build. If your file is purely typescript typing then it's preferred to have it as a d.ts file.

Reply:
I was wrong in this scenario. d.ts files are intended only as type declarations to corresponding .js files. So in this situation, the codegen should be creating .ts files. Will fix.

Reply:
Ahh, like in the case of retrofitting a package with a `@types/package`. That's where I'd seen `.d.ts` used for the most part; never really had a case for it in my own work so this is good to know! Thank you